modelci/app/v1/endpoints/model.py

```diff
@@ -35,3 +41,49 @@ def get_all_model(name: str = None, framework: Framework_ = None, engine: Engine
 def get_model(*, id: str):  # noqa
     model = ModelService.get_model_by_id(id)
     return ModelDetailOut.from_bo(model)
+
+
+@router.post('/', status_code=201)
+async def publish_model(
+        ml_model_in_form: MLModelInForm = Depends(MLModelInForm.as_form),
+        files: List[UploadFile] = File(
+            [],
+            description='This field can be set with empty value. In such settings, the publish is a dry run to'
+                        'validate the `ml_model_in_form` field. You are recommend to try a dry run before the'
+                        'model wight file(s) are posted. This is because the weight file(s) are usually large and '
+                        'take a while to receive the whole request and validate.'
+        ),
+        convert: bool = True,
+        profile: bool = False
+):
+    # save the posted files as local cache
+    loop = asyncio.get_event_loop()
+    saved_path = ml_model_in_form.saved_path
+    if len(files) == 0:
+        # dry run
```

Comment:
From definition
> A dry run is a testing process where the effects of a possible failure are intentionally mitigated.
> If a programmer found that the value is not what it should be, they are able to identify the section of code that resulted in the error.

In our case, this is for checking possible input errors in the request parameters. The consideration for this is when there is an error in the form data fields, the user will only know it after he uploads the files to the server. This takes me ~30s. 
So rather than let the user wait for 30s but prompt the input has errors, let them test input values without uploading the models first. If their input is correct, they can publish the model with the tested values.

Reply:
can we use a more human understandable manner?

I think few people will read the document so carefully and run dry run first.

My suggestion is to automate this part.

For instance, user only do once and the system uses two process to control this. 1. Send the input and check then sent back 2. If passed, upload the big file.

Reply:
This is not feasible for the backend, as the backend framework will process the one HTTP request as a whole. But our CLI can hide the dry run thing from the user perspective.